common/transport/amqp/src/amqp.ts

```diff
@@ -48,70 +123,85 @@ export class Amqp {
         False if the caller intends to manually settle messages
         A string containing the version of the SDK used for telemetry purposes] */
   constructor(autoSettleMessages: boolean) {
-    const autoSettleMode = autoSettleMessages ? amqp10.Constants.receiverSettleMode.autoSettle : amqp10.Constants.receiverSettleMode.settleOnDisposition;
+    const autoSettleMode = autoSettleMessages ? 1 : 2;
     // node-amqp10 has an automatic reconnection/link re-attach feature that is enabled by default.
     // In our case we want to control the reconnection flow ourselves, so we need to disable it.
 
-    /*Codes_SRS_NODE_COMMON_AMQP_16_042: [The Amqp constructor shall create a new `amqp10.Client` instance and configure it to:
+    /*Codes_SRS_NODE_COMMON_AMQP_16_042: [The Amqp constructor shall create a new `rhea.Client` instance and configure it to:
     - not reconnect on failure
-    - not reattach sender and receiver links on failure
-    - not reestablish sessions on failure]*/
-    this._amqp = new amqp10.Client(amqp10.Policy.merge(<any>{
-      session: {
-        reestablish: {
-          retries: 0,
-          forever: false
-        }
-      },
-      senderLink: {
-        attach: {
-          maxMessageSize: 0,
-        },
-        encoder: (body: any): any => {
-          if (typeof body === 'string') {
-            return new Buffer(body, 'utf8');
-          } else {
-            return body;
-          }
-        },
-        reattach: {
-          retries: 0,
-          forever: false
-        }
-      },
-      receiverLink: {
-        attach: {
-          maxMessageSize: 0,
-          receiverSettleMode: autoSettleMode,
-        },
-        decoder: (body: any): any => body,
-        reattach: {
-          retries: 0,
-          forever: false
-        }
-      },
-      // reconnections will be handled at the client level, not the transport level.
-      reconnect: {
-        retries: 0,
-        strategy: 'fibonacci',
-        forever: false
-      }
-    }, amqp10.Policy.EventHub));
+    - not reattach sender and receiver links on failure]*/
 
-    const amqpErrorHandler = (err) => {
-      debug('amqp10 client error: ' + err.toString());
+    this._rheaContainer = rheaCreateContainer();
+    this._rheaContainer.on('error', (context: EventContext) => {
+      debug('Container received a error event' + this._getErrorName(context.connection.error));
+    });
+
+    this._rheaContainer.on('azure-iot-amqp-base:error-indicated', (err: AmqpError) => {
```

Comment:
>this._rheaContainer.on('azure-iot-amqp-base:error-indicated' [](start = 4, length = 60)

this is a risky design pattern because we do not control the nature/type of the rhea container (what if it isn't an EventEmitter anymore?). it tells me we lack something in the current patterns that _we_ use. can we work on identifying the gap in our code, rather than rely on the modification of someone else's? or is rhea missing a thing and we need to ask for a change?

Reply:
To change the container or for that matter connections, links, sessions from event emitters would be a HUGE breaking architectural change for rhea.  Not. Going. To. Happen.

I need to emit an event up to this class.    This is what I have.  Give me an alternative and I'll consider.

---
In reply to: [209129983](https://github.com/Azure/azure-iot-sdk-node/pull/344#discussion_r209129983) [](ancestors = 209129983)